### PR TITLE
EA-168: Add methods to ExitFromCare service to support to marking a p…

### PR DESCRIPTION
**Pull request by mogoodrich**
…atient as "not dead"

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/emrapi/exitfromcare/ExitFromCareService.java [hunk 1]
```diff
@@ -30,14 +39,24 @@
...
-     * Marks the patient date with the specified cause of dead and death date
+     * Marks the patient dead with the specified cause of dead and death date
      * Closes any active programs or visits for that patient
      * TODO: better documentation here
      *
      * @param patient
      * @param causeOfDeath
      * @param deathDate
      */
-    void markPatientDied(Patient patient, Concept causeOfDeath, Date deathDate);
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/emrapi/exitfromcare/ExitFromCareService.java:41]: fyi, I tweaked the name of this method as "markPatientDead" seemed clearer... this method was just added in a recent commit and has not been released, so seemed okay to change.

File: api/src/main/java/org/openmrs/module/emrapi/exitfromcare/ExitFromCareService.java [hunk 2]
```diff
@@ -20,6 +20,15 @@
      */
     void closePatientPrograms(Patient patient, Concept outcome, Date date);
 
+    /**
+     * Reopens (ie, sets outcome and completion date to null) any programs that have an outcome equal to
+     * the outcome parameter, and a completion date equal to the completio date parameter
```
Comment (brandones) [inline api/src/main/java/org/openmrs/module/emrapi/exitfromcare/ExitFromCareService.java:25]:
```suggestion
     * the outcome parameter, and a completion date equal to the completion date parameter
```

File: api/src/main/java/org/openmrs/module/emrapi/exitfromcare/ExitFromCareService.java [hunk 3]
```diff
@@ -30,14 +39,24 @@
...
      * @param deathDate
      */
-    void markPatientDied(Patient patient, Concept causeOfDeath, Date deathDate);
+    void markPatientDead(Patient patient, Concept causeOfDeath, Date deathDate);
+
+
+    /**
+     * Removes the flag that set the patient as dead, and removes any cause of death
+     * Reopens any patient programs that have been closed with an outcome of "death" on the date of death
+     * TODO: better documentation here
```
Comment (brandones) [inline api/src/main/java/org/openmrs/module/emrapi/exitfromcare/ExitFromCareService.java:56]: > TODO: better documentation here

Any reason to put it off?

**Review by brandones — APPROVED** [review]
Very nice! Let's fill in the "TODO: better documentation here" now though, if you think it makes sense to do so.

File: api/src/main/java/org/openmrs/module/emrapi/exitfromcare/ExitFromCareService.java (hunk 3 above)
Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/emrapi/exitfromcare/ExitFromCareService.java:56]: Fair enough, I will flesh that out.

**Pull request merged**